ledger/src/shred.rs

```diff
@@ -467,8 +473,36 @@ impl Shred {
         self.common_header.signature
     }
 
-    pub fn seed(&self) -> [u8; 32] {
+    pub fn seed(
+        &self,
+        leader_schedule_cache: &Arc<LeaderScheduleCache>,
+        bank_forks: &Arc<RwLock<BankForks>>,
+    ) -> [u8; 32] {
         let mut seed = [0; 32];
+
+        let bf = bank_forks.read().unwrap();
+        let maybe_bank = bf.get(self.slot());
+        if let Some(bank) = maybe_bank {
+            if enable_deterministic_seed(self.slot(), bank) {
+                let maybe_leader_pubkey =
+                    leader_schedule_cache.slot_leader_at(self.slot(), Some(bank));
+                let h = match maybe_leader_pubkey {
+                    Some(pubkey) => hashv(&[
+                        &self.slot().to_le_bytes(),
+                        &self.index().to_le_bytes(),
+                        &pubkey.to_bytes(),
+                    ]),
+                    None => hashv(&[
+                        &self.slot().to_le_bytes(),
+                        &self.index().to_le_bytes(),
+                        self.common_header.signature.as_ref(),
+                    ]),
```

Comment:
This is changing the definition of seed without making it deterministic.
Probably better to resort back to old definition of seed here.

Reply:
changed